fda/functional_data.py

```diff
@@ -930,13 +930,11 @@ def plot(self, chart=None, *, derivative=0, fig=None, ax=None, nrows=None,
         else:
 
             if 'color' in kwargs:
-                sample_colors = np.asarray(
-                    kwargs.get("color")).repeat(self.nsamples)
+                sample_colors = self.nsamples * [kwargs.get("color")]
                 kwargs.pop('color')
 
             elif 'c' in kwargs:
-                sample_colors = np.asarray(
-                    kwargs.get("c")).repeat(self.nsamples)
+                sample_colors = self.nsamples * [kwargs.get("color")]
                 kwargs.pop('c')
 
             else:
```

Comment:
What about the else?

Reply:
I only made the minimal changes to fix the bug.
As far I know the else won't crash for the type of color. I think that this block of code should be rewrite using a generator and maybe moved to a plotting module.